Fix and modify the view of messages#index

queq1890 opened the pull request:
# WHAT
・メッセージ機能の実装に伴うビューの修正を行う。
(1)各グループの最新のコメントをサイドバーに表示する。
(2)ヘッダーに現在のグループ名、グループメンバーを表示する。
(3)ヘッダーのEDITボタンから各グループの編集画面に飛べるようにする。

# WHY
・アプリケーションを使いやすく、見やすくするため。

Commit:
first commit

Commit:
Add has_many :messages to user and group

Commit:
Show the latest message of each group on side bar

Commit:
Show group_name and group_members in index

Commit:
Show each group_name and path

Commit:
Fix url for message#index

Commit:
Set to conditions for what to show on sidebar as latest comment

Commit:
Expand the range of link_to ( now recent_comment is also clickable)

Commit:
Modify where to redirect after creating / editing group

queq1890 commented:
![mov](https://user-images.githubusercontent.com/32263803/31307367-250569f2-ab9e-11e7-99be-5059ff82826d.gif)
インデックスから編集画面に移動

queq1890 commented:
![mov](https://user-images.githubusercontent.com/32263803/31307371-406d62c6-ab9e-11e7-8a12-56e6213bdbb1.gif)
グループ名、メンバー名をクリックして対応するグループへ移動

queq1890 commented:
![mov](https://user-images.githubusercontent.com/32263803/31307377-53f49198-ab9e-11e7-9789-969f7b1be576.gif)
サイドバーに表示されるグループの最新のメッセージの動き

queq1890 commented:
<img width="1440" alt="2017-10-07 19 50 27" src="https://user-images.githubusercontent.com/32263803/31307382-7c71bb00-ab9e-11e7-83e1-3f5616eb30c5.png">

queq1890 commented:
<img width="1440" alt="2017-10-07 20 32 26" src="https://user-images.githubusercontent.com/32263803/31307394-b118f5c6-ab9e-11e7-94d2-1b804fa55bd3.png">
現在のmessages#indexのビュー

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -13,7 +14,7 @@ def create
       if @group.save
         flash[:notice] = "グループを作成しました"
-        redirect_to root_path
+        redirect_to "/groups/#{@group.id}/messages"
```
urlでなくpath表記にしましょう

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -25,7 +26,7 @@ def edit
       if @group.update(group_params)
         flash[:notice] = "グループを編集しました"
-        redirect_to root_path
+        redirect_to "/groups/#{@group.id}/messages"
```
こちらも

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -13,7 +14,7 @@ def create
-        redirect_to root_path
+        redirect_to "/groups/#{@group.id}/messages"
       else
         render 'new'
```
エラーメッセージを出しましょう

queq1890 replied:
<img width="1440" alt="2017-10-09 14 33 03" src="https://user-images.githubusercontent.com/32263803/31326052-c8528926-acfe-11e7-8e1c-e558c8378a0d.png">
現在、グループ作成時、編集時にエラーが起こった際はerrorsメソッドを使用してビューの方でエラーを表示しています。


```
  - if @group.errors.any?
    .alert.alert-danger{id:"error_explanation", class:"chat-group-form__errors"}
      %h2
        = "#{@group.errors.count}件のエラーが発生しました"
      %ul
        - @group.errors.full_messages.each do |message|
          %li
            = message
```

ここのレビューは、フラッシュメッセージでもエラーを表示した方が良いということでしょうか？

ta910 replied:
いえ、この記述で問題ないです。申し訳ありません。

ta910 commented on app/views/messages/index.html.haml:
```diff
@@ -3,12 +3,14 @@
-      グループ名
-    = link_to "Edit", "/", class:"right_contents__top__edit"
+      = @group.name
+    = link_to "Edit", "/groups/#{params[:group_id]}/edit", class:"right_contents__top__edit"
```
こちらもurlはpathにしましょう

ta910 commented on app/views/messages/index.html.haml:
```diff
@@ -3,12 +3,14 @@
+        = " #{user.name}"
   .right_contents__messages
-    - @messages.each do |message|
+    - @group.messages.each do |message|
```
メッセージはeachでなく部分テンプレートでまわしていただきたいです

ta910 commented on app/views/shared/_left_contents.html.haml:
```diff
@@ -8,5 +8,12 @@
     .left_contents__group
-      = group.name
-    .left_contents__group__recent_comment 最新のコメント
+      = link_to "/groups/#{group[:id]}/messages/", style:"text-decoration: none; color: #FFF;" do
```
こちらもpathで

ta910 commented on app/views/shared/_left_contents.html.haml:
```diff
@@ -8,5 +8,12 @@
 
   - current_user.groups.each do |group|
```
ここもeachでなく部分テンプレートにしましょう

Commit:
Modify path for group_controller

Commit:
Change path of views into prefix

Commit:
Create _message template and render it in messages/index.html.haml

Commit:
Add message.created_at  to _message.html.haml

Commit:
Make _left_contents__group.html.haml and render it in _left_contents.html.haml

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -13,7 +14,7 @@ def create
-        redirect_to root_path
+        redirect_to group_messages_path(@group)
       else
         render 'new'
```
エラーの時はエラーメッセージ出すようにしましょう

ta910 commented on app/controllers/groups_controller.rb:
```diff
@@ -25,7 +26,7 @@ def edit
-        redirect_to root_path
+        redirect_to group_messages_path(@group)
       else
         render 'edit'
```
ここもですね

ta910 commented on app/views/messages/index.html.haml:
```diff
@@ -3,22 +3,14 @@
-            = message.body
-          - else
-            = image_tag message.image
+    = render partial: 'message',collection: @group.messages
```
```render @group.messages```
でうまくいきませんでしょうか

ta910 commented on app/views/shared/_left_contents.html.haml:
```diff
@@ -6,7 +6,4 @@
-    .left_contents__group
-      = group.name
-    .left_contents__group__recent_comment 最新のコメント
+  = render partial: '/shared/left_contents__group', collection: current_user.groups, as: 'group'
```
こちらも部分テンプレート名を_group.html.hamlとし、
```render current_user.groups```
で記述できるかと思います

Commit:
Modify how to render right_contents__message

記述を短いものに変更

Commit:
Rename _group.html.haml into _group_form.html.haml

既にeditとnewでrenderされていた既存の_group.html.hamlの名前を変更

Commit:
Rename template and modify how to render _group.html.haml in _left_contents.html.haml

/shared/left_contents__group.html.hamlの名前を_group.html.hamlに変更、views/grou
psに移動、_left_contents.html.hamlでのレンダーの記述を簡略なものに変更

Commit:
Modify path to groups#edit

パスが間違っていたので修正

ReoKuriyama commented:
LGTM！
